File: frontend/src/app/core/services/espace-departement.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { Employe } from './employe.service';
import { Carte, Recharge, TypeCarteEmploye, StatutCarte } from './carte.service';

export interface EmployeDepartementRequest {
  nom: string;
  prenom: string;
  email: string;
  telephone?: string | null;
  matricule?: string | null;
}

export interface EspaceDepartementInfo {
  departementId: number;
  departementNom: string;
  entrepriseNom: string;
  budgetDisponible: number;
  nombreEmployes: number;
  cartesActives: number;
  cartesSuspendues: number;
  cartesBloquees: number;
  cartesExpirees: number;
  soldeTotalCartes: number;
}

@Injectable({ providedIn: 'root' })
export class EspaceDepartementService {
  private apiUrl = `${environment.apiUrl}/departement`;

  constructor(private http: HttpClient) {}

  getInfo(): Observable<EspaceDepartementInfo> {
    return this.http.get<EspaceDepartementInfo>(`${this.apiUrl}/info`);
  }

  getEmployes(): Observable<Employe[]> {
    return this.http.get<Employe[]>(`${this.apiUrl}/employes`);
  }

  createEmploye(request: EmployeDepartementRequest): Observable<Employe> {
    return this.http.post<Employe>(`${this.apiUrl}/employes`, request);
  }

  updateEmploye(id: number, request: EmployeDepartementRequest): Observable<Employe> {
    return this.http.put<Employe>(`${this.apiUrl}/employes/${id}`, request);
  }

  deleteEmploye(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/employes/${id}`);
  }

  getCartes(): Observable<Carte[]> {
    return this.http.get<Carte[]>(`${this.apiUrl}/cartes`);
  }

  createCarte(carte: Carte): Observable<Carte> {
    return this.http.post<Carte>(`${this.apiUrl}/cartes`, carte);
  }

  changerStatutCarte(id: number, statut: StatutCarte): Observable<Carte> {
    return this.http.patch<Carte>(`${this.apiUrl}/cartes/${id}/statut`, { statut });
  }

  rechargerCarte(id: number, montant: number): Observable<Recharge> {
    return this.http.post<Recharge>(`${this.apiUrl}/cartes/${id}/recharger`, { montant });
  }

  historiqueRecharges(id: number): Observable<Recharge[]> {
    return this.http.get<Recharge[]>(`${this.apiUrl}/cartes/${id}/recharges`);
  }
}

File: backend/src/main/java/cardoil/backend/service/EspaceDepartementService.java
package cardoil.backend.service;

import cardoil.backend.dto.request.CarteRequest;
import cardoil.backend.dto.request.EmployeDepartementRequest;
import cardoil.backend.dto.request.RechargeRequest;
import cardoil.backend.dto.response.CarteResponse;
import cardoil.backend.dto.response.EmployeResponse;
import cardoil.backend.dto.response.EspaceDepartementInfoResponse;
import cardoil.backend.dto.response.RechargeResponse;
import cardoil.backend.entity.StatutCarte;

import java.util.List;

public interface EspaceDepartementService {

    EspaceDepartementInfoResponse getInfo(String login);

    List<EmployeResponse> getEmployes(String login);
    EmployeResponse createEmploye(String login, EmployeDepartementRequest request);
    EmployeResponse updateEmploye(String login, Long employeId, EmployeDepartementRequest request);
    void deleteEmploye(String login, Long employeId);

    List<CarteResponse> getCartes(String login);
    CarteResponse createCarte(String login, CarteRequest request);
    CarteResponse changerStatutCarte(String login, Long carteId, StatutCarte statut);
    RechargeResponse rechargerCarte(String login, Long carteId, RechargeRequest request);
    List<RechargeResponse> getHistoriqueRecharges(String login, Long carteId);
}